packages/components/src/Badge/Badge.module.css

```
@@ -0,0 +1,101 @@
+.badge {
+  border-radius: var(--spacing-12);
+  color: var(--color-purple-800);
+  display: inline-block;
+  position: relative;
+  font-family: var(--typography-paragraph-extra-small-font-family);
+  font-size: var(--typography-paragraph-extra-small-font-size);
+  font-weight: var(--typography-paragraph-bold-font-weight);
+  letter-spacing: var(--typography-paragraph-extra-small-letter-spacing);
+  line-height: var(--typography-paragraph-extra-small-line-height);
+  padding: 1px var(--spacing-6);
+  min-width: 8px;
+  text-align: center;
+  background-color: var(--badge-background-color, var(--color-gray-300));
+
+  .reversed {
+    --badge-background-color: rgba(var(--color-white-rgb), 0.1);
+
+    color: var(--color-white);
+  }
+
+  .reversed.active {
+    --badge-background-color: var(--color-green-300);
+
+    color: var(--color-purple-800);
+  }
+
+  .reversed.dark {
+    --badge-background-color: var(--color-purple-700);
+
+    color: var(--color-white);
+  }
+}
+
+.badge.large {
+  display: inline-flex;
+  justify-content: center;
+  border-radius: var(--spacing-48);
+  font-size: var(--typography-data-medium-font-size);
+  line-height: var(--typography-data-medium-line-height);
+  letter-spacing: var(--typography-data-medium-letter-spacing);
+  padding: 2px 1.875rem;
+  width: 24px;
+}
+
+.default {
+  color: var(--color-purple-800);
+}
+
+.active {
+  --badge-background-color: var(--color-blue-500);
+
+  color: var(--color-white);
+}
```

Hopefully this helps a little in understanding the structure?

```suggestion
}

.reversed {
  --badge-background-color: rgba(var(--color-white-rgb), 0.1);

  color: var(--color-white);
}

// TODO: .dark.reversed moved into .dark

.large {
  display: inline-flex;
  justify-content: center;
  border-radius: var(--spacing-48);
  font-size: var(--typography-data-medium-font-size);
  line-height: var(--typography-data-medium-line-height);
  letter-spacing: var(--typography-data-medium-letter-spacing);
  padding: 2px 1.875rem;
  width: 24px;
  
  // TODO:
  // &.dot { ... } can probably go here, but it does mean that .dot must go above it.
  // this .large can likely go further down in the file after the variants
}

.default {
  color: var(--color-purple-800);
}

.active {
  --badge-background-color: var(--color-blue-500);

  color: var(--color-white);
  
  &.reversed {
    --badge-background-color: var(--color-green-300);

  color: var(--color-purple-800);
  }
}
```